tapis-project/tapis-ui
# task/TUI-64 - github pages publishing

## Overview: ##

Added npm script gh-pages that publishes a production version to https://tapis-project.github.io/tapis-ui/

## Related Github Issues: ##

* [TUI-64](https://github.com/tapis-project/tapis-ui/issues/64)

## Summary of Changes: ##

To publish, run 

```
npm run gh-pages
```

This will run a production build, commit the files to the gh-pages branch and push to github. The gh-pages branch will only be used to serve the application in this repo and should not be otherwise modified manually.

## Testing Steps: ##
1. Make some local change
2. Run `npm run gh-pages` to build and publish
3. Check https://tapis-project.github.io/tapis-ui/ to see your changes
4. Browse between sections of the site

## UI Photos:

## Notes: ##

The router has been changed to HashRouter, which eliminates some routing issues that come up using BrowserRouter. There is a way to remove the hashtag in the route, but that requires more configuration.
